src/libstore/globals.hh

```diff
@@ -1031,6 +1046,24 @@ public:
           ```
         )"
     };
+
+    EnvironmentSetting environment{this, {}, "environment",
```

Comment:
I'm not sure the name is right any more now that this setting doesn't blindly change the environment of the daemon.
Maybe something like `impure-env-var-overrides` to link it more explicitly to the `impureEnvVar` option? Probably that should be also reflected in the documentation, and there should be a reference from the documentation of `impureEnvVar` to this setting.